src/main/java/jp/co/future/uroborosql/filter/SecretColumnSqlFilter.java

```diff
@@ -171,8 +175,15 @@ public Parameter doParameter(final Parameter parameter) {
 					if (StringUtils.isNotEmpty(objStr)) {
 						try {
 							synchronized (encryptCipher) {
+								if (encryptCipher.getIV() != null) {
+									encryptCipher.init(Cipher.ENCRYPT_MODE, secretKey);
+								}
 								byte[] crypted = encryptCipher.doFinal(StringUtils.defaultString(objStr).getBytes(
 										getCharset()));
+								byte[] iv = encryptCipher.getIV();
```

Comment:
Is it better to obtain the local variable iv once at the beginning of syncronized?

Reply:
It is meaningful to get getIV after `Cipher.init`.